src/main/java/opendcs/dao/DaoBase.java

```diff
@@ -72,13 +102,26 @@ public void close()
 		if (queryStmt2 != null)
 			try { queryStmt2.close(); } catch(Exception ex) {}
 		queryStmt1 = queryStmt2 = null;
+		
+		// for pooling: return the connection (if there is one) back to the pool.
+		if (myCon != null && !conSetManually)
+			db.freeConnection(myCon);
+		myCon = null;
 	}
 	
 	public void finalize()
```

Comment:
How would try-with-resources alleviate the need for the close() method?
I don't use SonarQube, but in the CwmsTimeSeriesDb.getConnection/freeconnection methods I do print a log message with the number of outstanding open connections.
Things seem to work well in all the GUIs and utilities. I'm not 100% sure about compproc because the rma test database queue handler is not working. Is there any movement on that?

Reply:
By calling it automatically.

```java
try( Connecition conn = get...) {
   // dostuff
}
/*
 implicit here is a finally {
   conn.close();
}
block.
*/
```

Reply:
try-with-resources also null checks for you FYI